PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskSDKL.h

```diff
@@ -0,0 +1,93 @@
+#ifndef AliAnalysisTaskSDKL_H
+#define AliAnalysisTaskSDKL_H
+
+// $Id$
+
+class TH1;
+class TH2;
+class TH3;
+class THnSparse;
+//class TTree;
+class TNtuple;
+class AliJetContainer;
+class AliParticleContainer;
+
+//#include "AliAnalysisTaskEmcal.h"
+#include "AliAnalysisTaskEmcalJet.h"
+#include "FJ_includes.h"
+
+namespace fastjet {
+    class PseudoJet;
+}
+
+struct split {
+    float z;
+    float r;
+    float m;
+    int sd_step;
+};
+
+class AliAnalysisTaskSDKL : public AliAnalysisTaskEmcalJet {
+ public:
+
+  AliAnalysisTaskSDKL();
+  AliAnalysisTaskSDKL(const char *name, Int_t const backgroption);
+  virtual ~AliAnalysisTaskSDKL();
+
+  void                        UserCreateOutputObjects();
+  void                        Terminate(Option_t *option);
+
+  static AliAnalysisTaskSDKL* AddTaskSoftDrop(
+    const char *ntracks            = "usedefault",
+    const char *njets              = "Jets",
+    const char *nrho               = "Rho",
+    Int_t       nCentBins          = 1,
+    Double_t    jetradius          = 0.4,
+    Double_t    jetptcut           = 1,
+    Double_t    jetareacut         = 0.6,
+    const char *type               = "EMCAL",
+    Int_t       backgroption       = 0,
+    Int_t       leadhadtype        = 0,
+    const char *taskname           = "AliAnalysisTaskSDKL"
+  );
+
+  static std::vector<split>   ReclusterFindHardSplits(AliEmcalJet *jet);
+  static std::vector<split>   ReclusterFindHardSplits(fastjet::PseudoJet const & jet);
+  static std::vector<split>   ReclusterFindHardSplits(std::vector <fastjet::PseudoJet> const & particles);
+  static std::vector<split>   FindHardSplits(fastjet::PseudoJet const & jet);
```

Comment:
This will allow me to use them easily in e.g response task. Other way I see is to create separate class but it would be specific for my goals.

Reply:
Please have a look at PWGJE/EMCALJetTasks/AliLundPlaneHelper.h and check whether you can extend this or implement something similar. Otherwise you can use multiple inheritance - I use it in my tasks on data to share the trigger selection and it works fine (look at PWGJE/EMCALJetTasks/Tracks/AliAnalysisEmcalTriggerSelectionHelper.h and PWGJE/EMCALJetTasks/Tracks/AliAnalysisTaskEmcalJetEnergySpectrum.h). Accessing the whole analysis task only because of shareing functionality other tasks one should avoid.

Reply:
OK, amended in this task.